packages/orm/src/getProperties.ts

```diff
@@ -92,9 +92,20 @@ export function getProperties(meta: EntityMetadata): Record<string, any> {
     ),
   );
 
+  // Since our fake instance is actually generating the callbacks for our lazy fields, it will be captured in any
+  // lambdas created.  If any of them reference `this`, then they'll actually be referencing the fake instance.  So we
+  // need to clear out any properties directly on the fake instance now that we're done with it and use a proxy to
+  // intercept any attempts to access `this` from within the callbacks and fail.
+  Object.setPrototypeOf(instance, instancePrototypeProxy);
+  for (const prop of Object.getOwnPropertyNames(instance)) {
+    if (prop !== "__data") delete instance[prop];
+  }
+
   return cached;
 }
 
+const instancePrototypeProxy = new Proxy({}, { get: () => fail("Cannot use 'this' in a property callback") });
```

Comment:
Wouldn't mind renaming this to something that shows it is "only for the post-boot phase", like `disallowAfterBoot` / `throwAfterBoot` / I dunno, don't love those names, but something like that.

As-is, just from the name in isolation, it sounds like its always the fake instance's prototype